pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -65,400 +59,3 @@ func Convert_v1beta1_ScaleStatus_To_autoscaling_ScaleStatus(in *extensionsv1beta
-	out.ObjectMeta = in.ObjectMeta
-	return Convert_v1beta1_NetworkPolicySpec_To_networking_NetworkPolicySpec(&in.Spec, &out.Spec, s)
-}
-
-func Convert_networking_NetworkPolicy_To_v1beta1_NetworkPolicy(in *networking.NetworkPolicy, out *extensionsv1beta1.NetworkPolicy, s conversion.Scope) error {
-	out.ObjectMeta = in.ObjectMeta
-	return Convert_networking_NetworkPolicySpec_To_v1beta1_NetworkPolicySpec(&in.Spec, &out.Spec, s)
-}
-
-func Convert_v1beta1_NetworkPolicySpec_To_networking_NetworkPolicySpec(in *extensionsv1beta1.NetworkPolicySpec, out *networking.NetworkPolicySpec, s conversion.Scope) error {
-	if err := s.Convert(&in.PodSelector, &out.PodSelector, 0); err != nil {
-		return err
-	}
-	out.Ingress = make([]networking.NetworkPolicyIngressRule, len(in.Ingress))
-	for i := range in.Ingress {
-		if err := Convert_v1beta1_NetworkPolicyIngressRule_To_networking_NetworkPolicyIngressRule(&in.Ingress[i], &out.Ingress[i], s); err != nil {
-			return err
-		}
-	}
-	out.Egress = make([]networking.NetworkPolicyEgressRule, len(in.Egress))
-	for i := range in.Egress {
-		if err := Convert_v1beta1_NetworkPolicyEgressRule_To_networking_NetworkPolicyEgressRule(&in.Egress[i], &out.Egress[i], s); err != nil {
-			return err
-		}
-	}
-	if in.PolicyTypes != nil {
-		in, out := &in.PolicyTypes, &out.PolicyTypes
-		*out = make([]networking.PolicyType, len(*in))
-		for i := range *in {
-			if err := s.Convert(&(*in)[i], &(*out)[i], 0); err != nil {
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-func Convert_networking_NetworkPolicySpec_To_v1beta1_NetworkPolicySpec(in *networking.NetworkPolicySpec, out *extensionsv1beta1.NetworkPolicySpec, s conversion.Scope) error {
-	if err := s.Convert(&in.PodSelector, &out.PodSelector, 0); err != nil {
-		return err
-	}
-	out.Ingress = make([]extensionsv1beta1.NetworkPolicyIngressRule, len(in.Ingress))
-	for i := range in.Ingress {
-		if err := Convert_networking_NetworkPolicyIngressRule_To_v1beta1_NetworkPolicyIngressRule(&in.Ingress[i], &out.Ingress[i], s); err != nil {
-			return err
-		}
-	}
-	out.Egress = make([]extensionsv1beta1.NetworkPolicyEgressRule, len(in.Egress))
-	for i := range in.Egress {
-		if err := Convert_networking_NetworkPolicyEgressRule_To_v1beta1_NetworkPolicyEgressRule(&in.Egress[i], &out.Egress[i], s); err != nil {
-			return err
-		}
-	}
-	if in.PolicyTypes != nil {
-		in, out := &in.PolicyTypes, &out.PolicyTypes
-		*out = make([]extensionsv1beta1.PolicyType, len(*in))
-		for i := range *in {
-			if err := s.Convert(&(*in)[i], &(*out)[i], 0); err != nil {
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-func Convert_v1beta1_NetworkPolicyIngressRule_To_networking_NetworkPolicyIngressRule(in *extensionsv1beta1.NetworkPolicyIngressRule, out *networking.NetworkPolicyIngressRule, s conversion.Scope) error {
-	out.Ports = make([]networking.NetworkPolicyPort, len(in.Ports))
```

Comment:
This results in ports always being non-nil. The auto-generated method preserves nil if the input is nil.

Reply:
I'd be in favor of using the auto-generated function and doing a single line of fixup with a comment about why we're preserving compatibility, e.g.:

```go
if err := autoConvert_v1beta1_NetworkPolicyIngressRule_To_networking_NetworkPolicyIngressRule(...); err != nil {
  return err
}
if out.Ports == nil {
  // produce a zero-length non-nil slice for compatibility with prior manual conversion
  out.Ports = make([]networking.NetworkPolicyPort, 0)
}
```

Reply:
Done